ssl/record/rec_layer_s3.c

```diff
@@ -1586,10 +1600,12 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
              * above.
              * No alert sent because we already sent close_notify
              */
+            SSL3_RECORD_set_length(rr, 0);
+            SSL3_RECORD_set_read(rr);
             SSLfatal(s, SSL_AD_NO_ALERT, SSL_F_SSL3_READ_BYTES,
                      SSL_R_APPLICATION_DATA_AFTER_CLOSE_NOTIFY);
+            return -1;
         }
-        return -1;
```

Comment:
It's unclear why we need to pull the ```return -1``` into both arms of the conditional.

Reply:
Because in the first arm there is a codepath that does not return -1 and we continue. In fact that is the whole point. If we receive KeyUpdate or NewSessionTicket we want to continue onwards to process those messages without returning with -1.